plugin/evm/orderbook/memory_database.go

```diff
@@ -272,22 +272,102 @@ func (db *InMemoryDatabase) LoadFromSnapshot(snapshot Snapshot) error {
 }
 
 // assumes that lock is held by the caller
-func (db *InMemoryDatabase) Accept(blockNumber uint64, blockTimestamp uint64) {
+func (db *InMemoryDatabase) Accept(acceptedBlockNumber, blockTimestamp uint64) {
 	db.mu.Lock()
 	defer db.mu.Unlock()
 
-	for orderId, order := range db.OrderMap {
-		lifecycle := order.getOrderStatus()
-		if (lifecycle.Status == FulFilled || lifecycle.Status == Cancelled) && lifecycle.BlockNumber <= blockNumber {
-			delete(db.OrderMap, orderId)
-			continue
+	count := db.configService.GetActiveMarketsCount()
+	for m := int64(0); m < count; m++ {
+		longOrders := db.getLongOrdersWithoutLock(Market(m), nil, nil, false)
+		shortOrders := db.getShortOrdersWithoutLock(Market(m), nil, nil, false)
+
+		for _, longOrder := range longOrders {
+			status := shouldRemove(acceptedBlockNumber, blockTimestamp, longOrder)
+			if status == CHECK_FOR_MATCHES {
```

Comment:
what happens if we don't check for matches at this point and keep that order in memory?

Reply:
discussed offline.